### ISSUE TITLE
Upgrading MvvmCross to 7.1.2: XF App throws exception on Android when using DatePicker

### ISSUE THREAD

**Issue by zoli13** [MvvmCross/MvvmCross#4011]
## 🔙 Regression

Upgrading MvvmCross to 7.1.2: XF App throws exception on Android when using DatePicker

### Current behavior
We used our XF app with MvvmCross 5.xx without problems.
We updated to 7.1.2 (and also XF to latest), and it throws on Android. 
iOS is still ok.

### Reproduction steps
Just call `picker.Focus();`
Exception: Android.Views.WindowManagerBadTokenException: 'Unable to add window -- token null is not valid; is your activity running?'

### Configuration

**Version:** 7.1.2

**Platform:** 
XF on Android

**Please note** 
It has been also reported to Xamarin Forms, where **additional info is also provided**:
https://github.com/xamarin/Xamarin.Forms/issues/8979

**Comment by Cheesebaron** [comment]
We don't have official support for latest Xamarin forms.

**Comment by zoli13** [comment]
> We don't have official support for latest Xamarin forms.

@Cheesebaron what is the latest Xamarin Forms you officially support (using 7.1.2) ?

**Comment by Cheesebaron** [comment]
Check the dependencies here: https://www.nuget.org/packages/MvvmCross.Forms/

**Comment by zoli13** [comment]
That dependencies list you wrote, it says Xamarin.Forms (>= 4.6.0.847)
So I downgraded my 4.8.x to 4.6.0.847.  MvvmCross 7.1.2.
And still, using DatePicker crashes (in some view cases, like we have a custom tab control), which worked well with older 5.x MvvmCross.
Issue is with Android Context.
But I can apply the workaround mentioned here: https://github.com/xamarin/Xamarin.Forms/issues/8979

But we just noticed another issue.

If entry is in listview (not sure of so simple gui layout is enough, will see later), and clicking on that, it crashes (nullreference). It also seems to related to Android Context, as unhandled exception comes from: https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Platform.Android/ViewRenderer.cs [line 113]:
`Window window = Context.GetActivity().Window;`

Attached the full callstack. 
[stack.txt](https://github.com/MvvmCross/MvvmCross/files/5672278/stack.txt)

I don't see how can I workaround here. Any suggestion?
Seems to be similar to this one: https://github.com/MvvmCross/MvvmCross/issues/2832
This has a workaround suggestion, but not detailed how/where to apply.
